src/components/TodoAddForm/TodoAddForm.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+
+import { PreparedTodo, User } from '../../types';
+
+type Props = {
+  users: User[],
+  todos: PreparedTodo[],
+  addTask: (newTask: PreparedTodo) => void,
+};
+
+const generateNewId = (tasks: PreparedTodo[]) => {
+  const tasksIds = tasks.map(({ id }) => id);
+
+  return Math.max(...tasksIds) + 1;
+};
+
+export const TodoAddForm: React.FC<Props> = ({ users, todos, addTask }) => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setUserError(false);
+  };
+
+  const resetForm = () => {
+    setTitle('');
+    setTitleError(false);
+    setSelectedUserId(0);
+    setUserError(false);
+  };
```

Comment:
You don`t call this function if your setTitleError or setUserError is true you can delete them from here

Reply:
From one point you are right, but from another - as long as it is _reset_ form function - I believe it is good, that it puts the entire state to some initial value